### PR TITLE
RPRBLND-1845: Update to latest USD and HdRPR

### PR DESCRIPTION
**Pull request by bnagirniak**
### PURPOSE
New USD release 21.08 is available and there are updates in HdRPR.

### EFFECT OF CHANGE
Updated USD to 21.08 release. Updated HdRPR.

### TECHNICAL STEPS
1. Updated submodules USD, HdRPR.
2. Created USD.patch which adjusts USD build for HdRPR. Without it HdRPR doesn't build.
3. Adjusted builds: build_usd.py, create_libs.py.
4. Removed hasattr(os, 'add_dll_directory') in engine/\_\_init__.py as not needed.
5. Improvement in build process: added -clean flag to tools/build.py

### CODE REVIEW

File: tools/data/USD.patch [hunk 1]
```diff
@@ -0,0 +1,88 @@
+diff --git a/pxr/imaging/hioOpenVDB/CMakeLists.txt b/pxr/imaging/hioOpenVDB/CMakeLists.txt
```
Comment (bsavery) [inline tools/data/USD.patch:1]: So my only question here is about this patch file and why its necessary.  Could we merge these changes or change hdRPR to allow?

Comment (bnagirniak) [reply tools/data/USD.patch:1]: There is a bug in USD with hioOpenVDB project. It was created without public interface, which produces build error in HdRPR. This patch fixes it. We also can provide this fix to USD project, create PR.

Comment (bsavery) [reply tools/data/USD.patch:1]: Please do.

Comment (bnagirniak) [reply tools/data/USD.patch:13]: Discussed with @hshakula, he will do such PR to USD project.

Comment (hshakula) [reply tools/data/USD.patch:13]: https://github.com/PixarAnimationStudios/USD/pull/1574
https://github.com/PixarAnimationStudios/USD/pull/1575

**Pull request merged**